operator/src/main/java/oracle/kubernetes/operator/helpers/WebhookHelper.java

```diff
@@ -177,28 +178,41 @@ public NextAction onSuccess(Packet packet, CallResponse<V1ValidatingWebhookConfi
         V1ValidatingWebhookConfiguration existingWebhookConfig = callResponse.getResult();
         if (existingWebhookConfig == null) {
           return doNext(createValidatingWebhookConfiguration(getNext()), packet);
-        } else if (shouldUpdate(existingWebhookConfig, model)) {
+        } else if (shouldUpdate(existingWebhookConfig)) {
           return doNext(replaceValidatingWebhookConfiguration(getNext(), existingWebhookConfig), packet);
         } else {
           return doNext(packet);
         }
       }
 
-      private boolean shouldUpdate(V1ValidatingWebhookConfiguration existingWebhookConfig,
-                                   V1ValidatingWebhookConfiguration model) {
-        return !getServiceNamespaceFromConfig(existingWebhookConfig).equals(getServiceNamespaceFromConfig(model));
+      private boolean shouldUpdate(V1ValidatingWebhookConfiguration existingWebhookConfig) {
+        return !getServiceNamespace(existingWebhookConfig).equals(getServiceNamespace(model))
+            || !Arrays.equals(getClientConfigCaBundle(existingWebhookConfig), getClientConfigCaBundle(model));
```

Comment:
Instead of checking individual fields of the Webhook configuration, I wonder if it's easier to just compare the entire existing webhook configuration with the new configuration. For e.g. something like -
return !existingWebhookConfig.equals(getFirstWebhook(model));

Reply:
That would be easier, but unfortunately, we cannot do that here. The existing object that we get back from the ApiServer may have additional fields that the locally-created model does not have, for example, the creation time. We just need to compare the two fields that we do care: the namespace and caBundle.

Reply:
My understanding is that equals method only considers apiVersion, kind, metadata and webhook specs and excludes the fields such as creation time.

```
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    } else if (o != null && this.getClass() == o.getClass()) {
      V1ValidatingWebhookConfiguration v1ValidatingWebhookConfiguration = (V1ValidatingWebhookConfiguration)o;
      return Objects.equals(this.apiVersion, v1ValidatingWebhookConfiguration.apiVersion) && Objects.equals(this.kind, v1ValidatingWebhookConfiguration.kind) && Objects.equals(this.metadata, v1ValidatingWebhookConfiguration.metadata) && Objects.equals(this.webhooks, v1ValidatingWebhookConfiguration.webhooks);
    } else {
      return false;
    }
  }
```

Reply:
The creation time is part of the metadata.

Reply:
I see, ok.